sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveStrategies.scala

```diff
@@ -170,6 +171,79 @@ object HiveAnalysis extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * When scanning or writing to non-partitioned Metastore Parquet tables, convert them to Parquet
+ * data source relations for better performance.
+ */
+case class ParquetConversions(sparkSession: SparkSession) extends Rule[LogicalPlan] {
```

Comment:
How about we pass the `conf` & the `HiveSessionCatalog` as parameters? This should be doable with the builder, and saves you from doing casting later-on